SpotifySearch/SpotifySearch/DetailViewController.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  DetailViewController.swift
+//  SpotifySearch
+//
+//  Created by Harichandan Singh on 10/28/16.
+//  Copyright © 2016 Harichandan Singh. All rights reserved.
+//
+
+import UIKit
+
+class DetailViewController: UIViewController {
+
+    @IBOutlet weak var fullSizeImageView: UIImageView!
+    var fullSizeImageString: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        APIRequestManager.manager.getData(apiEndpoint: fullSizeImageString!) { (data: Data?) in
+            if let d = data {
+                self.fullSizeImageView.image = UIImage(data: d)
+            }
+            DispatchQueue.main.async {
+                self.fullSizeImageView.reloadInputViews()
```

Comment:
Again, UI should be set on the main thread. 

` self.fullSizeImageView.image = UIImage(data: d)` is outside the DispatchQueue.main.async{} closure